<!--
Copyright (c) 2023 Analog Devices, Inc. All Rights Reserved.
This software is proprietary to Analog Devices, Inc. and its licensors.
-->
 
<!-- -*- mode: XML -*- -->
<launch>
 
  <!-- Arguments on name space and sensor name in gazebo -->
  <arg name="ns_prefix" default="tmcm1" />
 
  <!-- All arguments below belongs to tf frame -->
  <arg name="tmcl_base_name" default="tmcm0930" />
 
  <!-- "period" specifies how often to send a transform-->
  <arg name="period" default="100"/>
 
  <!-- TMCL Board frame
    change values of:
    * "board_parent_frame" to specify parent frame of tmcl board (setting to "map" as default)
    * "pos_x,y,z" for translation
    * "pos_qx,qy,qz,qw" for rotation[Quaternion] -->
  <arg name = "board_parent_frame" value = "map" />
  <arg name = "board_base_name" value = "$(eval arg('ns_prefix') + '/' + arg('tmcl_base_name') + '_base_link')" />
  <arg name = "board_pos_x" value = "0" />
  <arg name = "board_pos_y" value = "0" />
  <arg name = "board_pos_z" value = "0" />
  <arg name = "board_pos_qx" value = "0" />
  <arg name = "board_pos_qy" value = "0" />
  <arg name = "board_pos_qz" value = "0" />
  <arg name = "board_pos_qw" value = "1" />
 
  <group ns="$(arg ns_prefix)">
    <!--Launches node-->
    <node name="tmcl_ros_node" pkg="adi_tmcl" type="tmcl_ros_node" output="screen" required="true">
      <!-- Autogenerated YAML file containing TMCM-0930 configurations -->
      <rosparam command="load" file="$(find adi_tmcl)/config/autogenerated/TMCM-0930.yaml" />
      <!-- User-generated YAML file containing ROS-specific parameters as well as user-set values for TMCM-0930 configurations -->
      <rosparam command="load" file="$(find adi_tmcl)/config/TMCM-0930_Ext.yaml" />
    </node>
 
    <!--Launches tf frames-->
    <node pkg="tf" type="static_transform_publisher" name="$(eval arg('tmcl_base_name') + '_link')" 
      args="$(arg board_pos_x) $(arg board_pos_y) $(arg board_pos_z) $(arg board_pos_qx) $(arg board_pos_qy) $(arg board_pos_qz) $(arg board_pos_qw) $(arg board_parent_frame) $(arg board_base_name) $(arg period)" />
  </group>
 
</launch>
